lib/rex/ui/text/shell/history_manager.rb

```diff
@@ -0,0 +1,68 @@
+# -*- coding: binary -*-
+
+module Rex
+module Ui
+module Text
+module Shell
+
+class HistoryManager
+ 
+
+  @@contexts = [{"history_file" => Msf::Config.history_file, "name" => :msfconsole}]
+  
+  def self.inspect
+    "#<HistoryManager stack size: #{@@contexts.length}>"
+  end
+
+  def self.context_stack
+    @@contexts
+  end
+
+  def self.push_context(history_file: nil, name: nil)
+    return if @@contexts[-1]['name'] == name
+    self.clear_readline
+    @@contexts.push({"history_file" => history_file, "name" => name})
+    if history_file
+      self.set_history_file(history_file)
+    end
+  end
+
+  def self.pop_context
+    if @@contexts.empty?
+      return
+    end
+    history_file, name = @@contexts.pop.values
+    if history_file
+      cmds = []
+      history_diff = Readline::HISTORY.size - @@original_histsize
+      history_diff.times do 
+        cmds.push(Readline::HISTORY.pop)
+      end
+      File.open(history_file, "a+") { |f| 
+        f.puts(cmds.reverse) }
+    end
+    self.clear_readline
+  end
+
+
+  def self.set_history_file(history_file)
```

Comment:
Can you make this a private method?

Reply:
okay :+1:

Reply:
done :+1:

Reply:
okay :+1:

Reply:
Thanks for the link, I've updated this :+1: